cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/batch/source/FileBatchSource.java

```diff
@@ -86,12 +87,15 @@
   private static final String TABLE_DESCRIPTION = "Name of the Table that keeps track of the last time files " +
     "were read in.";
   private static final String INPUT_FORMAT_CLASS_DESCRIPTION = "Name of the input format class, which must be a " +
-    "subclass of FileInputFormat. Defaults to TextInputFormat";
+    "subclass of FileInputFormat. Defaults to CombineTextInputFormat";
   private static final String FILESYSTEM_DESCRIPTION = "Distributed file system to read in from.";
+  private static final String MAX_SPLIT_SIZE_DESCRIPTION = "Max split size for each mapper in the MapReduce Job. " +
+    "Defaults to 100KB.";
   private static final Gson GSON = new Gson();
   private static final Logger LOG = LoggerFactory.getLogger(FileBatchSource.class);
   private static final Type ARRAYLIST_DATE_TYPE  = new TypeToken<ArrayList<Date>>() { }.getType();
   private static final Type MAP_STRING_STRING_TYPE = new TypeToken<Map<String, String>>() { }.getType();
+  private static final int DEFAULT_SPLIT_SIZE = 102400;
```

Comment:
According to [this|https://github.com/apache/hadoop/blob/trunk/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/main/java/org/apache/hadoop/mapreduce/lib/input/FileInputFormat.java#L216], the default for the param ``mapreduce.input.fileinputformat.split.maxsize`` is ``Long.MAX_VALUE``. I think for a realistic usecase, the default you have here is very low. We should probably have it atleast as the default size of 1 block on HDFS - 64 (now 128 in most cases) MB.